docs/architecture.md

```diff
@@ -2,32 +2,76 @@
+
+### Action
+
+Actions are tiny `struct`s or `enum`s that contain declarative language about either the triggering user action or the update request for a given `Store`.
+
+### Dispatcher
+
+The dispatcher class is the simplest in the application; it provides an `Action`-accepting method as a wrapper for the `PublishSubject<Action>` that publishes all dispatched actions to interested `Stores`:
+
+```
+class Dispatcher {
+    static let shared = Dispatcher()
+    private let storeDispatchSubject = PublishSubject<Action>()
+
+    open var register: Observable<Action> {
+        return self.storeDispatchSubject.asObservable()
+    }
+
+    open func dispatch(action: Action) {
+        self.storeDispatchSubject.onNext(action)
+    }
+}
+```
+
+### Store
+
+Stores provide an opaque wrapper around system storage or simple `Replay- /Publish- Subject`s for the purposes of data access and view configuration.
+
+### View Routing
 
-### Usage
+The special case in this scenario is view routing. To handle the view-changing component of the architecture, there is a `RouteStore` observed by a `RootPresenter` that rides along on the back of a `RootView`. This “containing” view will never be displayed to the user; rather, it will perform the role of listening for navigation-specific `Action`s & performing the necessary top-level navigation stack swapping or navigation stack manipulation. Routing logic lives entirely separately from individual view configuration logic, allowing for modular view manipulation and easy testing.
 
-RxSwift observables are used as the hooks between the dispatcher and stores, the stores & views, and in Action-specific cases, to deal with asynchronous application requirements.
+### Example
 
-The View component of Flux is represented by the Presenter+View couples. The Presenter is responsible for listening to the appropriate Store observables, and firing Actions to their respective handlers.
+To fully understand the concept, it's useful to trace one user action through its lifecycle of use in the app. Following is a simplified description of how the filter field (or search box) on the main item list screen works.
 
-### Routing
+When a user enters text into the search field, the textfield binding[2] on the `ItemListView` emits an event to an observer on the `ItemListPresenter`. The `ItemListPresenter` dispatches a `ItemListFilterAction`, which is a simple struct with one property - `filteringText: String`. The struct does a round trip through the `ItemListDisplayActionHandler`, `Dispatcher`, and `ItemListDisplayStore` before getting combined with the most recent list of items. This combined `Observable` stream with both the text and the items filters the list of items and maps the filtered list into individual cell configurations. The view, on receiving the updated / filtered list, re-renders the list of items to only show the ones that the user is searching for.
```

Comment:
I think it would help to break this down into an ordered list; I suggest start by making each sentence a list item.